src/test/static/class/class-properties.js

```diff
@@ -0,0 +1,25 @@
+class Class {
+  /** @public a static property with a getter */
+  static get customStaticPropertyGetter() {
+    return true;
+  }
+
+  /** @public a property with a getter */
+  get customPropertyGetter() { return true; }
+
+  /** @public a property with a getter/setter */
+  get customPropertyWithGetterSetter() {
+    return this._customPropertyWithGetterSetter;
+  }
+
+  set customPropertyWithGetterSetter(val) {
```

Comment:
We should also test the case where the setter comes before the getter. I think your code handles this correctly now, but it would be an easy thing for us to accidentally break later.

Reply:
yup i've added that now